lib/feature/one_hot_encoder.cc

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+
+#include "clink/feature/one_hot_encoder.h"
+#include "nlohmann/json.hpp"
+
+using namespace clink;
+using namespace std;
+using namespace nlohmann;
+
+SparseVector OneHotEncoderModel::transform(int x, int columnIndex) {
+  int len = modelData_.featuresize(columnIndex);
+  if (getDropLast()) {
+    len -= 1;
+  }
+
+  SparseVector vector(len);
+  if (x < len) {
+    vector.set(x, 1.0);
+  }
+  return vector;
+}
+
+void OneHotEncoderModel::setDropLast(bool isDropLast) {
+  params_.isDropLast = isDropLast;
+}
+
+bool OneHotEncoderModel::getDropLast() { return params_.isDropLast; }
+
+void OneHotEncoderModel::setModelData(std::string modelDataProtobuf) {
+  if (!modelData_.ParseFromString(modelDataProtobuf)) {
+    cerr << "Failed to parse modeldata." << endl;
```

Comment:
How about we use `TFRT_LOG(ERROR)` here? In the long term we can use INFO/WARNING/ERROR/FATAL log level similar to what we have with log4j in Java.

Same for other logging statements.

Reply:
So far all such error message can be returned by `llvm::Error` or `llvm::Expected`. I'll use these instead.